float32 x_position
float32 y_position
float32 heading_self
